clang/lib/Sema/HLSLExternalSemaSource.cpp

```diff
@@ -189,12 +189,28 @@ struct BuiltinTypeDeclBuilder {
   BuiltinTypeDeclBuilder &addArraySubscriptOperators(Sema &S) {
     if (Record->isCompleteDefinition())
       return *this;
-    addArraySubscriptOperator(S, true);
-    addArraySubscriptOperator(S, false);
+    ASTContext &AST = Record->getASTContext();
+    DeclarationName Subscript =
+        AST.DeclarationNames.getCXXOperatorName(OO_Subscript);
+    addHandleAccessFunction(S, Subscript, /*IsConst=*/true, /*IsRef=*/true);
+    addHandleAccessFunction(S, Subscript, /*IsConst=*/false, /*IsRef=*/true);
+    return *this;
+  }
+
+  BuiltinTypeDeclBuilder &addLoadMethods(Sema &S) {
```

Comment:
1. The name `addLoadMethods` implies that it will add multiple methods yet it is adding a single one.
2. What about Load with the `out uint status` parameter?
3. There are many other `Load` methods for different kinds of buffers/textures. Should the name reflect that?

Reply:
It's named `addLoadMethods` because I intend it to add the rest of the load methods (notably, the one with the `status` parameter). That method will look differently to the one we're adding here though, as it will need to call a builtin that does the load directly and also returns the status bit.

I hope that we can add the other load methods here too, but if not we can change the naming as needed.

Reply:
Sounds good. Maybe add a FIXME comment that the other Load methods will be added later?